cashier/src/main/java/com/getkeepsafe/cashier/Cashier.java

```diff
@@ -167,10 +170,20 @@ public void purchase(final Activity activity,
                        final PurchaseListener listener) {
     Preconditions.checkNotNull(product, "Product is null");
     Preconditions.checkNotNull(listener, "PurchaseListener is null");
+    if (Looper.myLooper() != Looper.getMainLooper()) {
+      throw new RuntimeException("purchase should only be called from the UI thread");
```

Comment:
Please improve the error message so it's clear that it comes from Cashier. something like 

`[Cashier] Cashier.purchase() should only be called from the UI thread`

Please also add a test case that checks for the exception. 
Also you might want to create a `CalledFromWrongTreadException` that extends a `RuntimeException` similar to `com.andorid.ViewImpl.CalledFromWrongTreadException`